src/Parser/Mage/Fire/Modules/Features/CastEfficiency.js

```diff
@@ -0,0 +1,192 @@
+import SPELLS from 'common/SPELLS';
+
+import CoreCastEfficiency from 'Parser/Core/Modules/CastEfficiency';
+
+/* eslint-disable no-unused-vars */
+
+class CastEfficiency extends CoreCastEfficiency {
+  static CPM_ABILITIES = [
+    ...CoreCastEfficiency.CPM_ABILITIES,
+    // Rotational spells
+    {
+      spell: SPELLS.FIREBALL,
+      category: CastEfficiency.SPELL_CATEGORIES.ROTATIONAL,
+      getCooldown: haste => null,
+    },
+    {
+      spell: SPELLS.PYROBLAST,
+      category: CastEfficiency.SPELL_CATEGORIES.ROTATIONAL,
+      getCooldown: haste => null,
+    },
+    {
+      spell: SPELLS.PHOENIXS_FLAMES,
+      category: CastEfficiency.SPELL_CATEGORIES.ROTATIONAL,
+      getCooldown: haste => 45,
+      recommendedCastEfficiency: 0.90,
+      charges: 3,
+    },
+    {
+      spell: SPELLS.FIRE_BLAST,
+      category: CastEfficiency.SPELL_CATEGORIES.ROTATIONAL,
+      getCooldown: haste => 12 / (1 + haste),
+      recommendedCastEfficiency: 0.90,
+      charges: 2,
+      isActive: combatant => !combatant.hasTalent(SPELLS.FLAME_ON_TALENT.id),
+    },
+    {
+      spell: SPELLS.FIRE_BLAST,
+      category: CastEfficiency.SPELL_CATEGORIES.ROTATIONAL,
+      getCooldown: haste => 10 / (1 + haste),
+      recommendedCastEfficiency: 0.90,
+      charges: 3,
+      isActive: combatant => combatant.hasTalent(SPELLS.FLAME_ON_TALENT.id),
```

Comment:
I think you also need to check for Soul of the Archmage too. (http://www.wowhead.com/item=151642/soul-of-the-archmage)

Reply:
That grants you the actual talent so it's enough to check for whether you have it.

Reply:
To be sure, have you verified with your code against actual logs?